### PR TITLE
CORE-9 Add Swagger docs to filesystem ticket endpoints.

### PR DESCRIPTION
**Pull request by psarando**
This PR will add the Swagger docs from cyverse-de/common-swagger-api#49 to the filesystem ticket endpoints.

### CODE REVIEW

File: src/terrain/routes/filesystem/tickets.clj [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  (optional-routes
+    [config/filesystem-routes-enabled]
+
+    (context
+      "/filesystem" []
+      :tags ["filesystem"]
+
+      (POST "/tickets" []
+            :middleware [coerce-public-param]
```
Comment (psarando) [inline src/terrain/routes/filesystem/tickets.clj]: The `public` param used to be optional, defaulting to a `false` value passed along to the required `data-info` param. It also used to only pass `true` to `data-info` if terrain's param was set to `1`.

This middleware is for backwards compatibility, and allows either `1` or `true` to set the `data-info` param to `true`, and anything else will pass along a `false`. The Swagger docs on this endpoint will only show `true` or `false` options, though.

File: src/terrain/routes/schemas/filesystem/tickets.clj [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            [schema.core :as s]
+            [schema-tools.core :as st]))
+
+;; Convert the keywords in the `mode` param to strings,
+;; so that the correct param format is passed through to the apps service.
+;; Remove `for-job` param.
+;; FIXME: Make `public` param optional (for backwards compatibility).
+(s/defschema AddTicketQueryParams
+  (-> schema/AddTicketQueryParams
+      (st/dissoc :for-job)
```
Comment (psarando) [inline src/terrain/routes/schemas/filesystem/tickets.clj]: I thought the `data-info` query param `for-job` was for internal use (by the `apps` service) only, so I'm not exposing it in the add or delete ticket endpoints in `terrain`.

Comment (slr71) [reply src/terrain/routes/schemas/filesystem/tickets.clj]: I _think_ that's accurate. We can always update it later if we need to.

File: src/terrain/routes/filesystem/tickets.clj [hunk 3]
```diff
@@ -0,0 +1,58 @@
+(ns terrain.routes.filesystem.tickets
```
Comment (slr71) [inline src/terrain/routes/filesystem/tickets.clj]: I like the idea of splitting this out into a separate namespace.

**Comment by psarando** [comment]
Thanks for the reviews! 👍

**Pull request merged**